### PR TITLE
v4.0.1 changelog

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,75 @@
...
+* [compiler] Escape properties in input builder function body (#6116)
+* [compiler] Provide a more descriptive error message when a resolution of a specific `ResolverKey` fails (#6136)
+* [compiler] Deprecate `@nonnull` (#6152)
+* [compiler] Allow mapping scalars to simple generic types (#6158)
+* [tooling] Allow to shutdown SchemaDownloader (#6091)
+* [tooling] Remove warning (#6092)
+* [WebSockets] connectionParams -> connectionPayload (#6103)
+* [WebSockets] add NetworkTransport.closeConnection() (#6105)
+* [runtime] Add 'OfflineException' as a cause when using failFastIfOffline (#6104)
+* [runtime] Change NetworkMonitor to expose a StateFlow directly (#6119)
```
Comment (martinbonnin) [inline CHANGELOG.md:56]: This is breaking and can probably be highlighted with a huge [BREAKING] or so?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,75 @@
...
+      }
+    }
+  }
+}
+```
+
+## ♻️ Deprecations
+- The `downloadApolloSchema` task is deprecated, please use an [`introspection` block](https://go.apollo.dev/ak-download-schema) to download a schema.
+- The `generateOperationOutput` property is deprecated, please use [`operationManifestFormat.set("persistedQueryManifest")`](https://www.apollographql.com/docs/kotlin/advanced/persisted-queries/#1-generate-operation-manifest) instead".
+- The `@nonnull` directive is deprecated, please use [`@semanticNonNull` / `@catch`](https://go.apollo.dev/ak-nullability) instead.
```
Comment (martinbonnin) [inline CHANGELOG.md:32]: Do we usually mention deprecations? I know some project do it (Gradle maybe?) but it's small signal to me as I get the information through other means (compiler warnings mostly).

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,75 @@
...
 ==========
 
-# Version v4.0.0
+# Version 4.0.1
+
+_2024-10-01_
+
+This release contains a handful of bug fixes and improvements. 
+
+## 🧩 Compiler Plugins additions (experimental)
```
Comment (martinbonnin) [inline CHANGELOG.md:10]:
```suggestion
## ⚙️ Add `ApolloCompilerPlugin.schemaListener()`
```

More specific title + my mind allocated 🧩 for the IJ plugin. Maybe not a huge deal but 🤷

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,75 @@
...
+  @ApolloExperimental
+  override fun schemaListener(): SchemaListener {
+    return object : SchemaListener {
+      override fun onSchema(schema: Schema, outputDirectory: File) {
+        // Generate some code inside outputDirectory based on schema
+      }
+    }
+  }
+}
+```
```
Comment (martinbonnin) [inline CHANGELOG.md:27]: If we're highlighting `ApolloCompilerPlugin.schemaListener`, we should also highlight generic support for scalars.

**Review by martinbonnin — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (BoD) [reply CHANGELOG.md:27]: I was hesitating but feel like it's more or less a bug fix 😅.

File: CHANGELOG.md (hunk 2 above)
Comment (BoD) [reply CHANGELOG.md:32]: Yeah we haven't done it before actually - ok to remove the section.

Comment (martinbonnin) [reply CHANGELOG.md:32]: Mind adding this rule to the CONTRIBUTING.md for reference?

**Pull request merged**